components/config/resources.rst

```diff
@@ -87,8 +87,16 @@ the resource::
     $loaderResolver = new LoaderResolver(array(new YamlUserLoader($locator)));
     $delegatingLoader = new DelegatingLoader($loaderResolver);
 
-    $delegatingLoader->load(__DIR__.'/users.yml');
-    /*
-    The YamlUserLoader will be used to load this resource,
+    // YamlUserLoader is used to load this resource because it supports
+    // files with the '.yml' extension
     since it supports files with a "yml" extension
-    */
+    $delegatingLoader->load(__DIR__.'/users.yml');
+
+.. tip::
+
+    If your application uses unconventional file extensions (for example, your
+    YAML files have a ``.res`` extension) you can pass the file type as the
```

Comment:
The only other loaders implementing `LoaderInterface` from the Config component are the routing ones.
They handle the `$type` argument quite differently:

```php

    /**
     * {@inheritdoc}
     */
    public function supports($resource, $type = null)
    {
        return is_string($resource) && in_array(pathinfo($resource, PATHINFO_EXTENSION), array('yml', 'yaml'), true) && (!$type || 'yaml' === $type);
    }
```

The type cannot be forced, but should be empty or match the extension. Maybe it can be updated to match the new behavior introduced in the di loaders.

But I still think this note should not appear in the Config component documentation, as it should described the `LoaderInterface` and what it is used for. It should not document implementations from other components.